src/buildercore/terraform.py

```diff
@@ -1020,7 +1020,7 @@ def _render_eks_workers_autoscaling_group(context, template):
 /etc/eks/bootstrap.sh --apiserver-endpoint '${aws_eks_cluster.main.endpoint}' --b64-cluster-ca '${aws_eks_cluster.main.certificate_authority.0.data}' '${aws_eks_cluster.main.name}'""")
 
     worker = {
-        'associate_public_ip_address': True,
+        'associate_public_ip_address': lookup(context, 'eks.worker.assign-public-ip', True),
```

Comment:
Thanks Luke. The default value for `lookup()` is documented as `0xDEADBEEF` - given that's never a valid value for this param, it felt like being explicit (in case there's a lack of default config or something) was better than running the risk of trying to throw an int at the AWS API somewhere down the line. 

You've raised the point about cfngen before though, and I think I keep missing it because so much of the terraform structure is currently checked and initialised here in `terraform.py`. I'll try to be more mindful, but to get to adding this extra bit would have meant moving an awful lot of code up-stack, which feels like a mountain for me. Gotta start somewhere though I guess. 

Thanks for looking it over, I've merged your changes anyway, will probably give this another once over tomorrow and maybe give it a go on flux-test